frontend/components/Text/Text.js

```diff
@@ -0,0 +1,34 @@
+import React from "react";
+import styled from "@emotion/styled";
+import { typography } from "../../theme";
+
+const StyledText = styled.text`
+  ${({ fontFamily }) => `font-family: "${fontFamily}";`}
+  ${({ as, fontWeight }) =>
+    as || fontWeight
+      ? `font-weight: ${fontWeight ?? typography[as].fontWeight};`
+      : ""}
+  ${({ as, fontSize }) =>
+    as || fontSize ? `font-size: ${fontSize ?? typography[as].fontSize};` : ""}
+  ${({ fontStyle }) => (fontStyle ? `font-style: ${fontStyle};` : "")}
+  ${({ color }) => (color ? `color: ${color};` : "")}
+  ${({ textDecoration }) =>
+    textDecoration ? `text-decoration: ${textDecoration};` : ""}
+  ${({ lineHeight }) => (lineHeight ? `line-height: ${lineHeight};` : "")}
+  ${({ cursor }) => (cursor ? `cursor: ${cursor};` : "")}
+`;
+
+const Text = ({
+  children,
+  as = "p",
```

Comment:
Can you elaborate on what this props purpose is?

Reply:
The purpose of the children prop is to be able to pass in strings, React elements and other components to be rendered by, in this case, the Text component. This is a necessity because it is the only way to render any of the component's descendants.
The as prop will tell the Text component to render its wrapper element as an HTML paragraph tag or another user specified tag. I opted to use the as prop because I wanted different variations of text to correspond to different HTML text tags like the span tag or the div tag.